elastic/kibana
# Add UUID to RuleAction

Resolves: [#149587](https://github.com/elastic/kibana/issues/149587)

This PR intends to add a `uuid` field to the rule actions.

A migration script add a uuid to all the existing actions.

Create method of the rule_client (and create endpoint) accepts a rule.actions data without uuid and adds a new uuid to the all actions.

Update and bulkEdit methods of the rule_client (and update and bulk_edit endpoints) accepts rule.actions data with and without uuid. As a user can add a new action to an existing rule, UI should send the uuid of an existing action back then update and bulkEdit methods would add the uuid's to all the new actions. 

All the get methods return uuid in the actions.

Since we don't query by the uuid of an action, I marked actions as `dynamic: false` in the mappings so we don't need to add the uuid to the mappings.

I tried not to modify the legacy APIs, therefore i used some type castings there, but please let me know if they need to be modified as well.

## To verify: 

Create a rule with some actions and save. 
Then add some more actions and expect all of them to have an auto-generated uuid field and still work as they were.

## Note:
Initialy uuid in `RuleAction` was not optional but as it created tons of changes on types and tests we decided to add it optional. It is still required in `RawRuleAction` which reflects the action object in Kibana SO.
There are 3 APIs in the `rulesClient` that adds a uuid: create, update, bulkEdit. All of them generates a uuid if there isn't any in an action, and since there is amigration script to add uuid to the existing actions, all the actions would have a uuid...
